### PR TITLE
Make in-season and not-in-season cards different

### PR DESCRIPTION
**Pull request by xster**
ezgif com-video-to-gif

You'll have to exercise some artistic taste on top of this but it looks like this

Fixes https://github.com/flutter/flutter/issues/30737

Tangentially fixes https://github.com/flutter/flutter/issues/30511

### CODE REVIEW

File: veggieseasons/lib/screens/list.dart [hunk 1]
```diff
@@ -65,12 +69,14 @@ class ListScreen extends StatelessWidget {
...
         );
 
-        rows.addAll(_generateVeggieRows(appState.unavailableVeggies, prefs));
+        rows.addAll(_generateVeggieRows(appState.unavailableVeggies, prefs, inSeason: false));
 
         return DecoratedBox(
           decoration: BoxDecoration(color: Color(0xffffffff)),
-          child: ListView(
-            children: rows,
+          child: CupertinoScrollbar(
```
Comment (RedBrogdon) [inline veggieseasons/lib/screens/list.dart]: Any reason this isn't in the Cupertino Widget Gallery? I didn't know it existed.

https://flutter.dev/docs/development/ui/widgets/cupertino

File: veggieseasons/lib/widgets/veggie_card.dart [hunk 2]
```diff
@@ -119,10 +123,23 @@ class VeggieCard extends StatelessWidget {
...
+            label: 'A card background featuring ${veggie.name}',
+            child: Container(
+              height: isInSeason ? 350 : 150,
+              decoration: BoxDecoration(
+                image: DecorationImage(
+                  fit: BoxFit.cover,
+                  colorFilter: isInSeason
+                      ? null
+                      // 222222 is a random color that has low color saturation.
+                      : ColorFilter.mode(Color(0xFF222222), BlendMode.saturation),
```
Comment (RedBrogdon) [inline veggieseasons/lib/widgets/veggie_card.dart]: I've been sticking random colors like this in `styles.dart`. Actually, ColorFilter has a const constructor, so you can probably just put the whole filter in there.

File: veggieseasons/lib/screens/list.dart (hunk 1 above)
Comment (xster) [reply veggieseasons/lib/screens/list.dart]: Oops, I didn't mean to keep this in. We'd need to fix https://github.com/flutter/flutter/issues/13253 first. I'll add it to the catalog tomorrow though.

File: veggieseasons/lib/widgets/veggie_card.dart (hunk 2 above)
Comment (xster) [reply veggieseasons/lib/widgets/veggie_card.dart]: ah sure. Done

**Comment by xster** [comment]
> How does this affect later plans to use the veggie cards as heroes on the way to the detail screen? 

Heros can go from whatever dimension to whatever dimension showing whatever we want in the process. 

> Is there a way to reach into the cards and lerp the saturation as the transition progresses?

Yes, we'd have to do a simple framework tweak on DecorationImage (could be your PR :D) though maybe it'd be cool to leave it desaturated too for off-season veggies.

**Pull request merged**